src/swtpm/swtpm.c

```diff
@@ -231,11 +234,39 @@ int swtpm_main(int argc, char **argv, const char *prgname, const char *iface)
                         "Given file descriptor type is not supported.\n");
                 exit(1);
             }
-            mlp.flags |= MAIN_LOOP_FLAG_TERMINATE | MAIN_LOOP_FLAG_USE_FD;
+            mlp.flags |= MAIN_LOOP_FLAG_USE_FD |            /* Use this as client fd */
+                         MAIN_LOOP_FLAG_TERMINATE |         /* Terminate if requested */
+                         MAIN_LOOP_FLAG_KEEP_CONNECTION;    /* Keep connection till the client connection active */
```

Comment:
Without this fix, in case of '-f' swtpm server is exiting just ofter serving a single request, that makes overall the 'clinentfd' patch useless :(., In my opinion this is a bug from swtpm side. Shall i put this change into a separate commit?